user_tools/src/spark_rapids_pytools/cloud_api/databricks_azure_job.py

```diff
@@ -34,16 +34,21 @@ def get_account_name(cls, url: str):
     def _build_jvm_args(self):
         vm_args = super()._build_jvm_args()
         key = ''
+        account_name = ''
+
+        eventlogs = self.exec_ctxt.get_value('wrapperCtx', 'eventLogs')
+        if eventlogs:
+            account_name = self.get_account_name(eventlogs[0])
+
         if 'key' in self.exec_ctxt.platform.ctxt:
             key = self.exec_ctxt.platform.ctxt['key']
-        else:
-            eventlogs = self.exec_ctxt.get_value('wrapperCtx', 'eventLogs')
-            if eventlogs and len(eventlogs) > 0:
-                account_name = self.get_account_name(eventlogs[0])
-                cmd_args = ['az storage account show-connection-string', '--name', account_name]
-                std_out = self.exec_ctxt.platform.cli.run_sys_cmd(cmd_args)
-                conn_str = JSONPropertiesContainer(prop_arg=std_out, file_load=False).get_value('connectionString')
-                key = conn_str.split('AccountKey=')[1].split(';')[0]
-        if key != '':
-            vm_args.append(f'-Drapids.tools.hadoop.fs.azure.account.key.databricksazuretest.dfs.core.windows.net={key}')
+        elif eventlogs:
```

Comment:
styling issue: `key` is very common name to be used to retrieve a key-value pair. It will be nice to rename that to something that reflects what the value it represents.

In this context, it looks like `self.exec_ctxt.platform.ctxt['key']`  has precedence on extracting account-name from eventlogs.
What sets `self.exec_ctxt.platform.ctxt['key']`? is that an input from the user?

Reply:
Yes, I agree. In azure cli, we can set the access key using `az config set storage.key=xxxxxx`. In the config file `~/.azure/config`, it would be under section `storage`, and key = `key`. I used 'key' mostly to conform with this azure cli convention.
If the user has set the access key using the azure cli command as above, then in `PlatformBase.load_from_config_parser` called by `PlatformBase._set_credential_properties` , this storage 'key' key-value pair would be picked up from the config file and stored in `self.exec_ctxt.platform.ctxt`.

Reply:
> Yes, I agree. In azure cli, we can set the access key using `az config set storage.key=xxxxxx`. In the config file `~/.azure/config`, it would be under section `storage`, and key = `key`. I used 'key' mostly to conform with this azure cli convention. If the user has set the access key using the azure cli command as above, then in `PlatformBase.load_from_config_parser` called by `PlatformBase._set_credential_properties` , this storage 'key' key-value pair would be picked up from the config file and stored in `self.exec_ctxt.platform.ctxt`.

I wonder if we need to use both methods in the wrapper.

Let's think about the following scenario:
- a user installs Azure SDK and he runs `az config set storage.key=xxxxxx` to set her/his key for a storage-account "STORAGE-A"
- User starts to explore our wrapper. It has been a while that she/he forgets that `az config set storage.key=xxxxxx` was executed.
- Because  she/he been using the Azure SDK for sometime, she/he thinks no need to change anything to the configurations configurations.
- she/he runs the wrapper command; but the eventlog is stored on another "STORAGE-B"
- The wrapper loads the key defined in the Azure config file to authenticate .
- The Scala module fails because it tries to authenticate "STORAGE-B" using "STORAGE-A" key.
- It won't be easy to troubleshoot the problem because we redact the actual key value. 

On the other hand, if we rely on a single source to load the key  which is the connection-string, then we are confident that we are loading the correct key for the eventlog (assuming they are all belong to same account-storage). WDYT?

Reply:
I think relying on a single source would make it easier for the user, given the scenario you pointed out above. So the user don't need to worry about setting the configuration or looking for the key in the Azure portal. I will update the implementation and test this.